optuna/importance/_base.py

```diff
@@ -63,43 +73,91 @@ def evaluate(
         raise NotImplementedError
 
 
-def _get_distributions(study: Study, params: Optional[List[str]]) -> Dict[str, BaseDistribution]:
+def _get_info_distribution(distribution: BaseDistribution) -> Dict[str, Any]:
+    info_distribution = {}
+    if isinstance(distribution, CategoricalDistribution):
+        info_distribution = {
+            "is_categorical": True,
+            "distribution": distribution,
+        }
+    elif isinstance(
+        distribution,
+        (
+            DiscreteUniformDistribution,
+            IntLogUniformDistribution,
+            IntUniformDistribution,
+            LogUniformDistribution,
+            UniformDistribution,
+        ),
+    ):
+        info_distribution = {
+            "is_categorical": False,
+            "low": distribution.low,
+            "high": distribution.high,
+        }
+
+    return info_distribution
+
+
+def _get_distributions(study: "Study", params: Optional[List[str]]) -> Dict[str, BaseDistribution]:
     _check_evaluate_args(study, params)
-
-    if params is None:
-        return intersection_search_space(study, ordered_dict=True)
-
-    # New temporary required to pass mypy. Seems like a bug.
-    params_not_none = params
-    assert params_not_none is not None
-
-    # Compute the search space based on the subset of trials containing all parameters.
-    distributions = None
-    for trial in study.trials:
+    info_distributions = None
+    for trial in reversed(study._storage.get_all_trials(study._study_id, deepcopy=False)):
```

Comment:
You might have reversed the iteration based on `IntersectionSearchSpace` but we don't a cursor here and don't think we can perform any  early breaks. Can we iterate over it in sequence for simplicity?